processing/src/main/java/io/druid/query/aggregation/hyperloglog/HyperLogLogCollector.java

```diff
@@ -92,8 +92,23 @@ public static HyperLogLogCollector makeLatestCollector()
 
   public static HyperLogLogCollector makeCollector(ByteBuffer buffer)
   {
+    return makeCollector(buffer, false);
+  }
+
+  /**
+   * Create a wrapper object around an HLL sketch contained within a buffer.
+   *
+   * @param buffer      buffer containing HLL sketch
+   * @param bufferOwned if true, we are permitted to change the position and limit of the buffer
```

Comment:
How do you feel about changing the contract of `makeCollector` to assume the buffer is owned (i.e. callers need to duplicate if they want to continue using the buffer)? I think having two methods is confusing, and I can just update all the call sites.